### DISCUSSION TITLE
Runs

### DISCUSSION THREAD

**Discussion by igboyes** [virtool/.github#7] [Ideas]
Add the concept of a run to Virtool.

| Field | Description |
|------|--------|
| `id` | Virtool ID for the run | 
| `created_at` | when the run was created in Virtool |
| `started_at` | when the run started
| `completed_at` | when the run completed
| `instrument` | the instrument (ID) that generated the run

# Features
- run-level QC (see Multi-QC)
- integrate with virtool instrument watcher service
- run ownership

**Comment by MikeERott** [comment]
Other fields could be, the name of the analyst, research project ID that the samples are for, whether these  are research or diagnostic samples.
It would also be nice to have a field to indicate which samples were grouped during the library prep and which were grouped during extraction/collection step. This might be easy once we have the robotics in place and could use run numbers from these machines. Distinguishing between robotic vs hand done libraries/ extractions would be good.
The feature being that by understanding the groupings of extractions vs libraries vs sequencing and correlating this data with common virus species detection in these samples, we can do a better job of accounting for cross contamination.

**Reply by igboyes** [reply]
> name of the analyst

This means a list of analysts should also be maintained in the Virtool database

> research project ID

Should probably be linked to a Virtool project and that represents the research project (**added**)

> research or diagnostic

Should probably extends labels feature to other entities other than samples. This would fulfill the requirement. You could make labels for _Research_ and _Diagnostics_

> a field to indicate which samples were grouped during the library prep
> which were grouped during extraction/collection step

These concepts need a lot more thought about how it would be represented in Virtool and automated. Manually entering these numbers/IDs would be very error prone. May be good idea to have something called _Run Group_.
